ext/libdatadog_api/datadog_ruby_common.c

```diff
@@ -47,11 +47,20 @@ ddog_prof_Endpoint endpoint_from(VALUE exporter_configuration) {
     VALUE site = rb_ary_entry(exporter_configuration, 1);
     VALUE api_key = rb_ary_entry(exporter_configuration, 2);
 
-    return ddog_prof_Endpoint_agentless(char_slice_from_ruby_string(site), char_slice_from_ruby_string(api_key));
+    ddog_Endpoint *endpoint = NULL;
+    ddog_Error *error = ddog_endpoint_from_api_key_and_site(
+        char_slice_from_ruby_string(site), char_slice_from_ruby_string(api_key),
+        &endpoint);
+    if (error != NULL) {
+      rb_raise(rb_eRuntimeError,
+               "Failed to initialize agentless endpoint: %" PRIsVALUE,
+               get_error_details_and_drop(error));
+    }
+    return endpoint;
```

Comment:
Minor: TBH I'm not sure if we care about agentless error tracking (E.g. if we're going to have a version of this function for crashtracking only, maybe we can just simplify and drop the agentless support there)

Reply:
Done!